File: backend/src/main/java/com/jugglehive/backend/model/entity/ttrpg/Chara.java
package com.jugglehive.backend.model.entity.ttrpg;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;

import com.jugglehive.backend.model.entity.login.User;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity(name = "chara")
@Data
@Table(schema = "ttrpg")
public class Chara {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id", nullable = false)
  private Long id;

  @OneToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "user_id", nullable = false)
  private User user;

  @OneToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "info_id", nullable = false)
  private CharacterInfo info;

  @Column(name = "age", nullable = false)
  private Integer age;

  @Column(name = "name", nullable = false, length = 50)
  private String name;

  @ManyToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "region_id", nullable = false)
  private Region region;

  @ManyToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "race_id", nullable = false)
  private Race race;

  @ManyToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "lvl_up_stat_id", nullable = false)
  private BaseStats lvlUpStat;

  @OneToMany(mappedBy = "chara")
  private List<CharacterClasses> classes = new ArrayList<>();


}

File: JuggleHiveWebapp.Server/Models/Chara.cs
namespace JuggleHiveWebapp.Server.Models;

public partial class Chara
{
    public long Id { get; set; }

    public long UserId { get; set; }

    public long InfoId { get; set; }

    public int Age { get; set; }

    public string Name { get; set; } = null!;

    public long RegionId { get; set; }

    public long RaceId { get; set; }

    public long LvlUpStatId { get; set; }

    public virtual ICollection<CharacterClass> CharacterClasses { get; set; } = new List<CharacterClass>();

    public virtual ICollection<CharacterSkill> CharacterSkills { get; set; } = new List<CharacterSkill>();

    public virtual ICollection<CharactersTreePoint> CharactersTreePoints { get; set; } = new List<CharactersTreePoint>();

    public virtual CharacterInfo Info { get; set; } = null!;

    public virtual ICollection<Inventory> Inventories { get; set; } = new List<Inventory>();

    public virtual BaseStat LvlUpStat { get; set; } = null!;

    public virtual Race Race { get; set; } = null!;

    public virtual Region Region { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}